install/update_0905_91.php

```diff
@@ -190,75 +196,106 @@ function update0905to91() {
    // cron task
    if (!countElementsInTable('glpi_crontasks',
                              ['itemtype' => 'ObjectLock', 'name' => 'unlockobject'])) {
-      $query = "INSERT INTO `glpi_crontasks`
-                       (`itemtype`, `name`, `frequency`, `param`, `state`, `mode`, `allowmode`,
-                        `hourmin`, `hourmax`, `logs_lifetime`, `lastrun`, `lastcode`, `comment`)
-                VALUES ('ObjectLock', 'unlockobject', 86400, 4, 0, 1, 3,
-                        0, 24, 30, NULL, NULL, NULL); ";
-      $DB->queryOrDie($query, "9.1 Add UnlockObject cron task");
+      $DB->insertOrDie("glpi_crontasks", [
+            'itemtype'        => "ObjectLock",
+            'name'            => "unlockobject",
+            'frequency'       => 86400,
+            'param'           => 4,
+            'state'           => 0,
+            'mode'            => 1,
+            'allowmode'       => 3,
+            'hourmin'         => 0,
+            'hourmax'         => 24,
+            'logs_lifetime'   => 30,
+            'lastrun'         => null,
+            'lastcode'        => null,
+            'comment'         => null
+         ],
+         "9.1 Add UnlockObject cron task"
+      );
    }
    // notification template
-   $query = "SELECT *
-             FROM `glpi_notificationtemplates`
-             WHERE `itemtype` = 'ObjectLock'";
-
-   if ($result = $DB->query($query)) {
-      if ($DB->numrows($result) == 0) {
-         $query = "INSERT INTO `glpi_notificationtemplates`
-                          (`name`, `itemtype`, `date_mod`)
-                   VALUES ('Unlock Item request', 'ObjectLock', NOW())";
-         $DB->queryOrDie($query, "9.1 Add unlock request notification template");
+   $notificationtemplatesIterator = $DB->request([
+      'FROM'   => "glpi_notificationtemplates",
+      'WHERE'  => ['itemtype' => "ObjectLock"]
+   ]);
+
+   if ($notificationtemplatesIterator->valid()) {
```

Comment:
Is this the correct way to check that the query didn't fail but returned 0 rows ?

Reply:
Nope, this method check if we're on a rwo, which is not the case. You should just `count($iterator)`.